fxgl/src/main/kotlin/com/almasb/fxgl/dsl/components/view/GenericBarViewComponent.kt

```diff
@@ -0,0 +1,39 @@
+/*
+ * FXGL - JavaFX Game Library. The MIT License (MIT).
+ * Copyright (c) AlmasB (almaslvl@gmail.com).
+ * See LICENSE for details.
+ */
+
+package com.almasb.fxgl.dsl.components.view
+
+import com.almasb.fxgl.ui.Position
+import com.almasb.fxgl.ui.ProgressBar
+import javafx.beans.property.SimpleIntegerProperty
+import javafx.scene.paint.Color
+
+/**
+ * A general bar view component to display properties like health, mana, or energy.
+ * Can be fully customized using its [bar] and [maxValue] property.
+ * @author Marvin Buff (marvinbuff@hotmail.com)
+ */
+open class GenericBarViewComponent(x: Double, y: Double, valueProperty: SimpleIntegerProperty, color: Color) : ChildViewComponent(x, y, false) {
+
+    val bar = ProgressBar()
+    val maxValue = SimpleIntegerProperty(valueProperty.get())
```

Comment:
I think we should expose `bar`'s current value and max value to avoid duplicate properties. So, maybe:
```
fun valueProperty(): DoubleProperty = bar.valueProperty()
fun maxValueProperty(): DoubleProperty = bar.maxValueProperty()
```

Then, the following is not needed:
```
        bar.currentValueProperty().bind(
                valueProperty.divide(maxValue).multiply(100)
        )
```